src/cecs277TermProject/RoachMotel.java

```diff
@@ -48,21 +48,37 @@ private RoachMotel(String name, int capacity) throws FileNotFoundException {
 	}
 
 	// TODO: [Gust] Add a parameter for list of amenities
-	public String checkIn(RoachColony roachColony) {
+public String checkIn(RoachColony roachColony, ArrayList<AmenityType>amenities) {
 		String result = String.format("Checking in: %s\n", roachColony);
 		logger.log(String.format("Colony %1$s attempting to check in.", roachColony));
 		if (availableRooms.isEmpty()) {
 			logger.log(String.format("Colony %1$s failed to check in, no available rooms.", roachColony));
 			return result + "No available room.";
 		}
+		
 		Room room = availableRooms.poll();
+		Room roomAmenity = room ;
+		 for (AmenityType amenity : amenities) {
+		        if (amenity == AmenityType.FOODBAR) {
+		            roomAmenity = new RoomWithFoodBar(roomAmenity);
+		        } else if (amenity == AmenityType.SPA) {
+		            roomAmenity= new RoomWithSpa(roomAmenity);
+		        } else if (amenity == AmenityType.AUTO_REFILL) {
+		            roomAmenity = new RoomWithAutoRefill(room);
+		        } else if (amenity == AmenityType.SPRAY_RESISTANT_SHOWER) {
+		            roomAmenity = new RoomWithSprayResistantShower(roomAmenity);
+		        }
+		    }
+		double cost = roomAmenity.getCost();
 		occupiedRooms.put(roachColony, room);
```

Comment:
We should use `roomAmenity` here, otherwise we will lose the amenity info later at checkout.

Reply:
wouldn't it print the amenities when you print the occupied rooms?

Reply:
Yes, that's what we want, right?

Reply:
The formatting is just a bit odd since the amenities are inside occupied rooms{}. Will amend the changes though.

Reply:
Looking at the sample output in the prompt, I see that the professor does print out the room amenities for each occupied room. I'm not sure how the formatting looks, if you can, please copy paste it here so that we can take a look.

Reply:
Checking in: First colony
Successfully checking in Room Number 101, Food Bar AutoRefill, Spa  
Total Cost:140.0
Motel(name=The only roach motel, availableRooms=[102, 103, 104, 105, 106], occupiedRooms=**{First colony=101, Food Bar AutoRefill, Spa}**

Reply:
looking good to me :) unless you think otherwise